### PR TITLE
Fixed some compile warnings with Clang on Windows

### PR DESCRIPTION
**Pull request by sammyfreg**
Fixed some Clang compile warnings under Windows, on Docking branch

### CODE REVIEW

File: backends/imgui_impl_win32.cpp [hunk 1]
```diff
@@ -272,7 +272,7 @@ static void ImGui_ImplWin32_UpdateGamepads()
 
 static BOOL CALLBACK ImGui_ImplWin32_UpdateMonitors_EnumFunc(HMONITOR monitor, HDC, LPRECT, LPARAM)
 {
-    MONITORINFO info = { 0 };
+    MONITORINFO info = { 0, {0,0,0,0}, {0,0,0,0}, 0 };
     info.cbSize = sizeof(MONITORINFO);
```
Comment (ocornut) [inline backends/imgui_impl_win32.cpp:276]: What is the warning here? Would using = `{};` fixes it?

File: imgui_widgets.cpp [hunk 2]
```diff
@@ -7305,7 +7305,8 @@ static ImU32   ImGui::TabBarCalcTabID(ImGuiTabBar* tab_bar, const char* label, I
 {
     if (docked_window != NULL)
     {
-        IM_ASSERT(tab_bar->Flags & ImGuiTabBarFlags_DockNode);
+        IM_UNUSED(tab_bar);
+        IM_ASSERT(tab_bar->Flags & ImGuiTabBarFlags_DockNode);        
         ImGuiID id = ImHashStr(label);
```
Comment (ocornut) [inline imgui_widgets.cpp:7310]: As a minor detail (probably more important for other PR) try to configure your IDE to trim trailing spaces.

I do know that VS2019 doesn't honor editor-config exactly like specs requires it (only trim trailing spaces when pressing Enter) but shouldn't be affected here.

File: backends/imgui_impl_win32.cpp (hunk 1 above)
Comment (sammyfreg) [reply backends/imgui_impl_win32.cpp:276]: Under Clang / VisualStudio2019 / Warning 4
>1>..\..\backends\imgui_impl_win32.cpp(275,28): warning : missing field 'rcMonitor' initializer [-Wmissing-field-initializers]

`={};` also Works. Will change it to that.

**Comment by ocornut** [comment]
Merged, thank you Sammy!

**Pull request closed without merging**